agent/http_oss_test.go

```diff
@@ -111,3 +113,29 @@ func TestHTTPAPI_MethodNotAllowed_OSS(t *testing.T) {
 		}
 	}
 }
+
+func TestHTTPAPI_OptionMethod_OSS(t *testing.T) {
+	a := NewTestAgent(t.Name(), `acl_datacenter = "dc1"`)
+	a.Agent.LogWriter = logger.NewLogWriter(512)
+	defer a.Shutdown()
+
+	for _, tt := range expectedEndpoints {
```

Comment:
Given that allowedMethods is package local and visible to this test there's no need to create another map. instead, you could loop over the keys and values in allowedMethods making the same OPTIONS call and verifying the list of options returned matches the values in allowedMethods.

Reply:
The problem with that is that
 ```	{"OPTIONS,GET", "/v1/query/xxx/execute"},
	{"OPTIONS,GET", "/v1/query/xxx/explain"},
``` 
won't be tested, and it won't actually check that `allowedMethods` was correctly defined. Is that OK?

Reply:
Good point. However this test method also relies on `expectedEndpoints` being comprehensively defined, so won't catch things that we forget to add here as we add more end points. 

I suggest explicitly adding test cases for the special cases like /execute/ and /explain in this method after looping through allowedMethods. Also add a comment line for it explaining why the special case lines exist. 

Any future end points (not special cased) added to the http server will be tested automatically without us having to maintain two lists.